utils/testsuite/testsuite_skiplist.py

```diff
@@ -2051,7 +2051,6 @@
     "import-assertions",
     "json-superset",
     "new.target",
-    "regexp-match-indices",
```

Comment:
This is good to test to spec, but it should not be the only source of testing. Add some examples of using the `d` flag and making sure it's working in some basic tests cases in our lit tests. You can add these cases to `test/hermes/regexp.js`. For instance, I haven't had time to debug this further, but I am getting an error when running this code on top of your changes:

```
let re = /(?<keyword>let|const|var)\s+(?<id>[a-zA-Z_$][0-9a-zA-Z_$]*)/d;
let match = re.exec("let myVar");
print((match.indices.groups));
```
Something funny is happening here since if I instead change the line to the following:
```
print(JSON.stringify((match.indices.groups)));
```
everything works fine.

Reply:
Yes I was going to say that, this behavior also happens to `match.groups` because it's a null-prototype object just like `match.indices.groups` is.

This issue also happens in other engines if you do like this for example (although they'll work if you just `console.log` the variable without string conversion):

```javascript
let re = /(?<keyword>let|const|var)\s+(?<id>[a-zA-Z_$][0-9a-zA-Z_$]*)/d;
let match = re.exec("let myVar");
console.log('match.groups ' + match.groups); // Uncaught TypeError: Cannot convert object to primitive value
console.log('match.indices.groups ' + match.indices.groups); // Uncaught TypeError: Cannot convert object to primitive value
```

Some references:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#return_value
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object#null-prototype_objects

Reply:
About the testing, I'll implement some tests as requested 👍 . Should I implement tests for this "null-prototype object" case too?

Reply:
Yeah I don't think I would classify this as a bug. JSC and SpiderMonkey both exhibit the same behavior as Hermes in this case. You can test this by trying to print out the expression `Object.create(null)` in the different engines.

You should not write tests for this case.